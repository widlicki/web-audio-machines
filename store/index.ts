import { configureStore } from '@reduxjs/toolkit'
import playbackControlReducer from '../components/machines/PlaybackControl/playbackControlSlice'
import stepControlReducer from '../components/machines/StepControl/stepControlSlice'

export function makeStore() {
  return configureStore({
    reducer: {
      playbackControl: playbackControlReducer,
      stepControl: stepControlReducer,
    },
  })
}

const store = makeStore()

export default store

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch
