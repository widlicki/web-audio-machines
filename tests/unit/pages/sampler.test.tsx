import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import Sampler from '../../../pages/machines/sampler'
import { makeStore } from '../../../store'

jest.mock(
  '../../../components/machines/PlaybackControl/PlaybackControl',
  () => (props: any) => {
    // @ts-ignore: unknown JSX element
    return <playback-control {...props} />
  }
)

jest.mock(
  '../../../components/machines/StepControl/StepControl',
  () => (props: any) => {
    // @ts-ignore: unknown JSX element
    return <step-control {...props} />
  }
)

jest.mock(
  '../../../components/machines/StepSequencer/StepSequencer',
  () => (props: any) => {
    // @ts-ignore: unknown JSX element
    return <step-sequencer {...props} />
  }
)

let component: any
let store: any

const createConnectedComponent = () => {
  return (
    <Provider store={store}>
      <Sampler />
    </Provider>
  )
}

describe('Sampler', () => {
  beforeEach(() => {
    store = makeStore()
  })
  it('renders sampler page according to snapshot', () => {
    component = createConnectedComponent()
    const { container } = render(component)
    expect(container).toMatchSnapshot()
  })
})
