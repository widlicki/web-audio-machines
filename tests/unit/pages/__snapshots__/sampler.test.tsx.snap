// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sampler renders sampler page according to snapshot 1`] = `
<div>
  <div>
    <h1>
      Sampler
    </h1>
  </div>
</div>
`;
